src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/UseCountProperly.cs

```diff
@@ -228,10 +228,11 @@ static void AddIfNotNull(ImmutableHashSet<IMethodSymbol>.Builder set, IEnumerabl
                 operationKey = OperationEqualsInstance;
             }
             // Analyze argument operation, potentially 0.Equals(obj.Count()).
-            else if (parentOperation is IArgumentOperation argumentOperation)
+            else if (parentOperation is IArgumentOperation argumentOperation &&
+                argumentOperation.GetAncestor<IInvocationOperation>(OperationKind.Invocation) is { } invocation)
```

Comment:
This would not work correctly if say this is an argument passed to an IObjectCreationOperation, which in turn is passed as an argument to an invocation.

Reply:
We should instead directly check if parent is an invocation.

Reply:
You are right! Besides, as we are on an argument operation there is no need to walk-up multiple nodes.